dlt-connector/src/interactions/backendToDb/transaction/CreateTransactionRecipe.context.test.ts

```diff
@@ -0,0 +1,338 @@
+    expect(transaction).toMatchObject({
+      type: TransactionType.GRADIDO_CREATION,
+      protocolVersion: '3.3',
+      community: {
+        rootPubkey: keyPair.publicKey,
+        foreign: 0,
+        iotaTopic: topic,
+      },
+      signingAccount: {
+        derive2Pubkey: moderator.account.derive2Pubkey,
+      },
+      recipientAccount: {
+        derive2Pubkey: firstUser.account.derive2Pubkey,
+      },
+      amount: new Decimal(2000),
+      backendTransactions: [
+        {
+          typeId: InputTransactionType.CREATION,
+        },
+      ],
+    })
+
+    const body = TransactionBody.fromBodyBytes(transaction.bodyBytes)
+    // console.log(new TransactionBodyLoggingView(body))
+    expect(body.creation).toBeDefined()
+    if (!body.creation) throw new Error()
+    const bodyReceiverPubkey = Buffer.from(body.creation.recipient.pubkey)
+    expect(bodyReceiverPubkey.compare(firstUser.account.derive2Pubkey)).toBe(0)
+
+    expect(body).toMatchObject({
+      type: CrossGroupType.LOCAL,
+      creation: {
+        recipient: {
+          amount: '2000',
+        },
+      },
+    })
+  })
+
+  it('local send transaction', async () => {
+    const sendTransactionDraft = new TransactionDraft()
+    sendTransactionDraft.amount = new Decimal('100')
+    sendTransactionDraft.backendTransactionId = 2
+    sendTransactionDraft.createdAt = new Date().toISOString()
+    sendTransactionDraft.linkedUser = secondUser.identifier
+    sendTransactionDraft.user = firstUser.identifier
+    sendTransactionDraft.type = InputTransactionType.SEND
+    const context = new CreateTransactionRecipeContext(sendTransactionDraft)
+    await context.run()
+    const transaction = context.getTransactionRecipe()
+
+    // console.log(new TransactionLoggingView(transaction))
+    expect(transaction).toMatchObject({
+      type: TransactionType.GRADIDO_TRANSFER,
+      protocolVersion: '3.3',
+      community: {
+        rootPubkey: keyPair.publicKey,
+        foreign: 0,
+        iotaTopic: topic,
+      },
+      signingAccount: {
+        derive2Pubkey: firstUser.account.derive2Pubkey,
+      },
+      recipientAccount: {
+        derive2Pubkey: secondUser.account.derive2Pubkey,
+      },
+      amount: new Decimal(100),
+      backendTransactions: [
+        {
+          typeId: InputTransactionType.SEND,
+        },
+      ],
+    })
+
+    const body = TransactionBody.fromBodyBytes(transaction.bodyBytes)
+    // console.log(new TransactionBodyLoggingView(body))
+    expect(body.transfer).toBeDefined()
+    if (!body.transfer) throw new Error()
+    expect(Buffer.from(body.transfer.recipient).compare(secondUser.account.derive2Pubkey)).toBe(0)
+    expect(Buffer.from(body.transfer.sender.pubkey).compare(firstUser.account.derive2Pubkey)).toBe(
+      0,
+    )
+    expect(body).toMatchObject({
+      type: CrossGroupType.LOCAL,
+      transfer: {
+        sender: {
+          amount: '100',
+        },
+      },
+    })
+  })
```

Comment:
no, it trust the backend to get a send and recv transaction from it.
On checking the recv transaction on GradidoNode the send transaction will be checked also and if not found or invalid, the recv transaction is invalid. 
Maybe I add sometimes a reverse check. Maybe I can use here the same logic as for the deferred transfer.

Reply:
As i remember it will be ensured by using the semaphore in the transactions method `executeTransaction` for local transactions.
But for x-com-tx the outbound-part will only be written in the sender-community and the inbound-part will be written in the receiver-community.
In consequence there will be gaps in the tx-identifier-values.